addons/hr_attendance/models/hr_attendance.py

```diff
@@ -168,7 +173,7 @@ def _update_overtime(self, employee_attendance_dates=None):
             )[emp.resource_id.id]
             # Substract Global Leaves and Employee's Leaves
             leave_intervals = emp.resource_calendar_id._leave_intervals_batch(
-                start, stop, emp.resource_id, domain=self._get_overtime_leave_domain()
+                start, stop, emp.resource_id, domain=self._get_overtime_leave_domainV1([False, emp.company_id.id])
```

Comment:
Why not simply doing that:
```suggestion
            domain = expression.AND([
                self._get_overtime_leave_domain(),
                [('company_id', 'in', [False, emp.company_id])],
            ])
            leave_intervals = emp.resource_calendar_id._leave_intervals_batch(
                start, stop, emp.resource_id, domain=domain
```
instead of deprecating the method? 🤔

Reply:
it's just to make it clear that you need to use V1 as people may continue using  _get_overtime_leave_domain without noticing that they need to add this AND.

i can remove it if you want

Reply:
With your V1, we completely remove the customisation overriding that method and so, they will not really notice it if they did not write a test. 😬

Reply:
okay, i will apply your suggestion